src/test/scala/wdl4s/types/WdlPairTypeSpec.scala

```diff
@@ -96,7 +122,58 @@ class WdlPairTypeSpec extends FlatSpec with Matchers {
         |  }
         |}
       """.stripMargin
-    
+
     noException should be thrownBy WdlNamespaceWithWorkflow.load(wdl, Seq.empty)
   }
+
+  it should "coerce a JsArray into a WdlArray of WdlPairs" in {
+    val jsArray = JsArray(
+      JsObject("Left" -> JsString("a"), "Right" -> JsNumber(1)),
+      JsObject("Left" -> JsString("b"), "Right" -> JsNumber(2)),
+      JsObject("Left" -> JsString("c"), "Right" -> JsNumber(3))
+    )
+
+    WdlArrayType(WdlPairType(WdlStringType, WdlIntegerType)).coerceRawValue(jsArray) match {
+      case Success(array) => array shouldEqual arrayOfPairs
+      case Failure(f) => fail(s"exception while coercing JsObject: $f")
+    }
+  }
+
+  it should "coerce a complex JsArray into a WdlArray of WdlPairs of WdlArrays" in {
```

Comment:
ToL: Making it the outermost element an array just:
- Makes the test harder to read
- Tests that WdlArray is correctly delegating to us

So, I'd be just as confident in this test if it just coerced a single element.